python/dgl/contrib/graph_store.py

```diff
@@ -404,29 +475,28 @@ def __init__(self, graph_name, port):
 
         # Set the ndata and edata initializers.
         # so that when a new node/edge embedding is created, it'll be created on the server as well.
-        def node_initializer(name, arr):
-            shape = F.shape(arr)
-            dtype = np.dtype(F.dtype(arr)).name
-            self.proxy.init_ndata(name, shape, dtype)
+
+        # These two functions create initialized tensors on the server.
+        def node_initializer(init, shape, dtype, ctx, name):
+            init = self._init_manager.serialize(init)
+            dtype = np.dtype(dtype).name
+            self.proxy.init_ndata(init, name, shape, dtype)
             data = empty_shared_mem(_get_ndata_path(self._graph_name, name),
                                     False, shape, dtype)
             dlpack = data.to_dlpack()
-            arr1 = F.zerocopy_from_dlpack(dlpack)
-            arr1[:] = arr
-            return arr1
-        def edge_initializer(name, arr):
-            shape = F.shape(arr)
-            dtype = np.dtype(F.dtype(arr)).name
-            self.proxy.init_edata(name, shape, dtype)
+            return F.zerocopy_from_dlpack(dlpack)
+        def edge_initializer(init, shape, dtype, ctx, name):
+            init = self._init_manager.serialize(init)
+            dtype = np.dtype(dtype).name
+            self.proxy.init_edata(init, name, shape, dtype)
             data = empty_shared_mem(_get_edata_path(self._graph_name, name),
                                     False, shape, dtype)
             dlpack = data.to_dlpack()
-            arr1 = F.zerocopy_from_dlpack(dlpack)
-            arr1[:] = arr
-            return arr1
-        self._node_frame.set_remote_initializer(node_initializer)
-        self._edge_frame.set_remote_initializer(edge_initializer)
-        self._msg_frame.set_remote_initializer(edge_initializer)
+            return F.zerocopy_from_dlpack(dlpack)
+
+        self._node_frame.set_remote_init_builder(lambda init: partial(node_initializer, init))
+        self._edge_frame.set_remote_init_builder(lambda init: partial(edge_initializer, init))
+        self._msg_frame.set_remote_init_builder(lambda init: partial(edge_initializer, init))
```

Comment:
Do we need another `set_remote_init_builder`? I feel the existing `set_initializer` is enough. The codes below are from my understanding, may not directly match your implementation.
```python
proxy = ...
node_frame = Frame(...)
def remote_init_fn(shape, dtype, ctx, name):
  proxy.init_data('zero', shape, dtype, name)
  data = empty_shared_mem(...)  # btw, the name empty here is actually confusing, the content is actually *not* empty at all.
  return data
node_frame.set_initializer(remote_init_fn)
```

Reply:
We can't only use `set_initializer`. When creating a column, we may want to create it from the shared memory. In this case, we need to get a remote initializer. However, when creating initialized data for zero-degree vertices, we only need one created in the local memory.
In addition, the signatures for the two initializers are different. the remote version needs an argument "name".